src/geom/SpanSet.cc

```diff
@@ -0,0 +1,640 @@
+    int beginY = box.getMinY();
+    int endY = box.getMaxY();
+
+    int beginX = box.getMinX();
+    int endX = box.getMaxX();
+
+    for (int i = beginY; i < (endY+1); ++i) {
+        spanVector.push_back(Span(i, beginX, endX));
+    }
+}
+
+// Construct a SpanSet from a std vector by copying
+geom::SpanSet::SpanSet(const std::vector<Span> & vec, bool normalize): spanVector(vec) {
+    if (normalize) {
+        runNormalize();
+    }
+    initialize();
+}
+
+// Construct a SpanSet from a std vector by moving
+geom::SpanSet::SpanSet(std::vector<Span> && vec, bool normalize): spanVector(std::move(vec)) {
+    if (normalize) {
+        runNormalize();
+    }
+    initialize();
+}
+
+void geom::SpanSet::runNormalize() {
+    // Ensure the span set is sorted according to Span < operator
+    std::sort(spanVector.begin(), spanVector.end());
+
+    // Now that the span is sorted, overlapping spans should be combined
+    // Start from 1 as the comparison will always be with the previous element
+    for (auto iter = ++(spanVector.begin()); iter != spanVector.end(); iter++) {
+        if (spansContiguous(*std::prev(iter, 1), *iter)) {
+            // These spans overlap, create one new one, covering the whole expanse
+            *(iter-1) = Span((*(iter-1)).getY(),
+                             std::min((*(iter-1)).getMinX(), (*iter).getMinX()),
+                             std::max((*(iter-1)).getMaxX(), (*iter).getMaxX()));
+            // Erase the current Span, as it is now contained in the previous element
+            iter = spanVector.erase(iter);
+            // Move the iterator back one to make sure the element gets run in the loop
+            iter = iter - 1;
+        }
+    }
+}
+
+
+void geom::SpanSet::initialize() {
+    /* This function exists to handle common functionality for most of the constructors. It  calculates the
+     * bounding box for the spanset, and the area covered by the SpanSet
+     */
+
+    /* Because the array is sorted, the minimum and maximum values for Y will
+       be in the first and last elements, only need to find the min and max X
+       values */
+    int minX = spanVector[0].getMinX();
+    int maxX = spanVector[0].getMaxX();
+    area = 0;
+
+    for (const auto & span : spanVector) {
+        if (span.getMinX() < minX) {
+            minX = span.getMinX();
+        }
+        if (span.getMaxX() > maxX) {
+            maxX = span.getMaxX();
+        }
+        // Plus one, because end point is inclusive
+        area += span.getMaxX() - span.getMinX() + 1;
+    }
+    bBox = geom::Box2I(Point2I(minX, spanVector.front().getY()), Point2I(maxX, spanVector.back().getY()));
+}
+
+// Getter for the area property
+std::size_t geom::SpanSet::getArea() const {
+    return area;
+}
+
+// Getter for the bounding box of the SpanSet
+geom::Box2I geom::SpanSet::getBBox() const {
+    return bBox;
+}
+
+
+bool geom::SpanSet::isContiguous() const {
```

Comment:
Are you assuming here that spans from a single y value can only have two disjoint spans? I'm not sure this algorithm works when you have three, but maybe I'm not understanding it correctly.